lib/web_ui/lib/src/engine/html_image_codec.dart

```diff
@@ -129,13 +129,13 @@ class HtmlImage implements ui.Image {
   final int height;
 
   @override
-  Future<ByteData?> toByteData(
-      {ui.ImageByteFormat format = ui.ImageByteFormat.rawRgba}) {
-    return futurize((Callback<ByteData?> callback) {
-      return _toByteData(format.index, (Uint8List? encoded) {
-        callback(encoded?.buffer.asByteData());
-      });
-    });
+  Future<ByteData?> toByteData({ui.ImageByteFormat format = ui.ImageByteFormat.rawRgba}) {
+    if (imgElement.src?.startsWith('data:') == true) {
+      final data = UriData.fromUri(Uri.parse(imgElement.src!));
+      return Future.value(data.contentAsBytes().buffer.asByteData());
```

Comment:
This ignores the `format` parameter. It's not entirely clear to me what format the `data:` URI is in, I'm trying to test this and I'm only seeing regular URIs or blob URIs.

Reply:
Oops, this was some time ago already, we spent some with finalizing the PR, so the code itself is even older. Let me think. I seem to remember PNG was the only format available in the data URL, so it simply couldn't support anything else. The possibilities are:

* Only serve PNG, throw for the others,
* Always serve PNG, no matter what is asked for (current),
* Convert if something else than PNG is asked for (unlikely, I guess).